### DISCUSSION TITLE
Update Tiles on updating the State

### DISCUSSION THREAD

**Discussion by Haseeba393** [react-native-maps/react-native-maps#4493] [Feature requests]
I'm using Aeris Weather Tile server for showing tile layer (radar, clouds, and storm etc) using UrlTile. I'm saving tile server URL in a state, the url is like:
`url: `https://maps.aerisapi.com/${CONFIG.AERIS_CLIENT_ID}/radar/{z}/{x}/{y}/current.png``
and have a menu where we can select multiple layers (radar, clouds and attacks), so on selecting any layer the state will get updated with updated url, but UrlTile is not updating the layers on Map.

**Comment by Haseeba393** [comment]
Its really a urgent ask from you all, please help me around in this

**Comment by muzammilideofuzion** [comment]
Is there one who can help me out in this regard. To update any Tile layer we have to do a force re-render by changing the key of UrlTile/WmsTile. Due to this, there is a little flickering in the layers that it disappeared immediately and then take some time again to load, this is really a weird behavior, is it possible to just update the new layer, while the current remain there until the new one will load?

**Comment by Haseeba393** [comment]
@pedro @netshade please help me out in this regard

**Comment by AlejandroRM-DEV** [comment]
I had the same problem, I solved it by repeating code

`
{mapType.tile === 'map_carton dark mode_1' && (
   <UrlTile urlTemplate="http://....." />
)}
{mapType.tile === 'map_carton positron_2' && (
  <UrlTile urlTemplate="https://......" />
)}
`

**Reply by Haseeba393** [reply]
I know this way but in this way UrlTile doesn't have smooth transitions. One disappear immediately and other one will appear once the native module will load it.
